collect_app/src/main/java/org/odk/collect/android/widgets/ExStringWidget.java

```diff
@@ -185,57 +178,17 @@ public void cancelLongPress() {
 
     @Override
     public void onButtonClick(int buttonId) {
-        String exSpec = getFormEntryPrompt().getAppearanceHint().replaceFirst("^ex[:]", "");
-        final String intentName = ExternalAppsUtils.extractIntentName(exSpec);
-        final Map<String, String> exParams = ExternalAppsUtils.extractParameters(exSpec);
-        final String errorString;
-        String v = getFormEntryPrompt().getSpecialFormQuestionText("noAppErrorString");
-        errorString = (v != null) ? v : getContext().getString(R.string.no_app);
-
-        Intent i = new Intent(intentName);
-
-        // Use special "uri_data" key to set intent data. This must be done before checking if an
-        // activity is available to handle implicit intents.
-        if (exParams.containsKey(URI_KEY)) {
-            try {
-                String uriValue = (String) ExternalAppsUtils.getValueRepresentedBy(exParams.get(URI_KEY),
-                            getFormEntryPrompt().getIndex().getReference());
-                i.setData(Uri.parse(uriValue));
-                exParams.remove(URI_KEY);
-            } catch (XPathSyntaxException e) {
-                Timber.d(e);
-                onException(e.getMessage());
-            }
-        }
-
-        if (!activityAvailability.isActivityAvailable(i)) {
-            Intent launchIntent = Collect.getInstance().getPackageManager().getLaunchIntentForPackage(intentName);
-
-            if (launchIntent != null) {
-                // Make sure FLAG_ACTIVITY_NEW_TASK is not set because it doesn't work with startActivityForResult
-                launchIntent.setFlags(0);
-                i = launchIntent;
-            }
-        }
-
-        if (activityAvailability.isActivityAvailable(i)) {
-            try {
-                ExternalAppsUtils.populateParameters(i, exParams,
-                        getFormEntryPrompt().getIndex().getReference());
-
-                waitingForDataRegistry.waitForData(getFormEntryPrompt().getIndex());
-                // ACTION_SENDTO used for sending text messages or emails doesn't require any results
-                if (ACTION_SENDTO.equals(i.getAction())) {
-                    getContext().startActivity(i);
-                } else {
-                    fireActivity(i);
-                }
-            } catch (ExternalParamsException | ActivityNotFoundException e) {
-                Timber.d(e);
-                onException(e.getMessage());
+        waitingForDataRegistry.waitForData(getFormEntryPrompt().getIndex());
```

Comment:
> I think it doesn't actually matter in practice because nothing will read that it's waiting for data, is that right?

Yes you are right we don't need to care about registering such questions because even if we register a question and then something wrong happens it won't cause any problems.